internal/csi/driver/driver.go

```diff
@@ -477,7 +478,11 @@ func (d *Driver) generateRequest(meta metadata.Metadata) (*manager.CertificateRe
 		return nil, err
 	}
 
-	jwttoken, err := jwt.ParseSigned(token)
+	// The signature algorithms we pass to ParseSigned are a best-effort guess, since Kubernetes don't seem to document a
+	// list of what might be used.
+	// The list is essentially taken from algorithms mentioned in
+	// https://github.com/kubernetes/kubernetes/blob/2b025e645975d6d51bf38c008f972c632cf49657/pkg/serviceaccount/jwt.go
+	jwttoken, err := jwt.ParseSigned(token, []jose.SignatureAlgorithm{jose.ES256, jose.ES384, jose.ES512, jose.RS256})
```

Comment:
Can we just allow all assymetric algorithms?
```
	jose.RS256,
	jose.RS384,
	jose.RS512,
	jose.ES256,
	jose.ES384,
	jose.ES512,
	jose.PS256,
	jose.PS384,
	jose.PS512,
	jose.EdDSA,
```

Reply:
Aaah, prompted by this I found an actual list: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L229-L240

The only difference there is EdDSA isn't supported by the looks of it, so I'll use the contents of that list here and update the comment.